### PR TITLE
Make compatible with Docker

### PR DESCRIPTION
**Pull request by pschambacher**
:octopus::computer:

/cc @osheroff @bquorning @grosser 

### Description
Allow database.yml to use MYSQL_URL environment variable to find the database.
Bit of refactor of creation/deletion of databases while I was around.

### Steps to merge
* [ ] :+1: from the team

### Risks
* None (test only)

### CODE REVIEW

**Comment by bquorning** [comment]
:+1:

**Comment by pschambacher** [comment]
Would like another review.

File: test/helper.rb [hunk 1]
```diff
@@ -4,28 +4,23 @@
...
-config =  YAML::load(IO.read(File.dirname(__FILE__) + '/database.yml'))
-
-if ENV["BUNDLE_GEMFILE"] =~ /mysql2/
-  config.each do |k, v|
-    v['adapter'] = 'mysql2'
-  end
-end
+config_content = IO.read(File.dirname(__FILE__) + '/database.yml')
+config_content = ERB.new(config_content).result
+config =  YAML::load(config_content)
```
Comment (grosser) [inline test/helper.rb]: FYI prefer '.' for static method calls

File: test/helper.rb [hunk 2]
```diff
@@ -68,6 +63,28 @@ class Test5 < ActiveRecord::Base
...
+  def for_each_database
+    ActiveRecord::Base.configurations.each do |name, conf|
+      next if name =~ /not_there/
+      next if conf['username'] == 'travis'
+      yield(name, conf)
+    end
+  end
+
+  def run_mysql_command(conf, command)
+    if @mysql_command.nil?
```
Comment (grosser) [inline test/helper.rb]: `@mysql_command ||= begin .... end` might read a bit nicer

File: test/test_arhp.rb [hunk 3]
```diff
@@ -3,10 +3,15 @@
 class ActiveRecordHostPoolTest < MiniTest::Unit::TestCase
   include ARHPTestSetup
   def setup
+    arhp_drop_databases
     arhp_create_databases
     arhp_create_models
   end
 
+  def teardown
```
Comment (grosser) [inline test/test_arhp.rb]: I like having both at the top too :)

**Comment by grosser** [comment]
:+1:

File: test/test_arhp.rb (hunk 3 above)
Comment (pschambacher) [inline test/test_arhp.rb]: top or bottom I don't care but I like having them together :smile:

**Pull request merged**